src/stream/src/executor/aggregation/mod.rs

```diff
@@ -396,6 +397,37 @@ pub fn generate_column_descs(
 
     column_descs
 }
+
+/// Generate state table for agg executor.
+/// Relational pk = `table_desc.len` - 1.
+pub fn generate_state_table<S: StateStore>(
+    ks: Keyspace<S>,
+    agg_call: &AggCall,
+    group_keys: &[usize],
+    pk_indices: &[usize],
+    agg_schema: &Schema,
+    input_ref: &dyn Executor,
+) -> StateTable<S> {
+    let table_desc = generate_column_descs(agg_call, group_keys, pk_indices, agg_schema, input_ref);
+    // Always leave 1 space for agg call value.
+    let relational_pk_len = table_desc.len() - 1;
+    StateTable::new(
+        ks,
+        table_desc,
+        // Primary key do not includes group key.
+        vec![
+            // Now we only infer order type for min/max in a naive way.
+            if agg_call.kind == AggKind::Max {
+                OrderType::Descending
```

Comment:
Would you please help check how we encode stream pk previously? For max agg, did we also encode pk in descending way?

Reply:
You mean write batch/keyspace? I think after this PR, as long as we are using state table, we are encode pk in correct order way.